storage: TestFS failed

storage.TestFS [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/14462740?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/14462740?buildTab=artifacts#/) on release-23.2 @ [332001fd06506fd28f3e47159c2e44692bc6af54](https://github.com/cockroachdb/cockroach/commits/332001fd06506fd28f3e47159c2e44692bc6af54):

Fatal error:

```
panic: test timed out after 4m55s
running tests:
	TestFS (7s)

```
Stack: 

```
goroutine 3131 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2259 +0x320
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestFS
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/94ddc9c34349c78aeaa54dc5cac2306d/logTestFS1381360996
    test_log_scope.go:81: use -show-logs to present logs inline
    engine_test.go:1503: -- test log scope end --
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/storage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestFS.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-36838

--- a/pkg/storage/BUILD.bazel
+++ b/pkg/storage/BUILD.bazel
@@ -143,8 +143,8 @@ go_test(
     args = ["-test.timeout=295s"],
     data = glob(["testdata/**"]),
     embed = [":storage"],
-    exec_properties = {"Pool": "medium"},
-    shard_count = 2,
+    exec_properties = {"Pool": "large"},
+    shard_count = 4,
     deps = [
         "//pkg/base",
         "//pkg/clusterversion",

